geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedQueryRunner.scala

```diff
@@ -51,6 +51,8 @@ class MergedQueryRunner(ds: HasGeoMesaStats, stores: Seq[(Queryable, Option[Filt
       original: Query,
       explain: Explainer): CloseableIterator[SimpleFeature] = {
 
+    // TODO deduplicate arrow, bin, density queries...
```

Comment:
Ok, how would updates come to the Lambda DataStore?  Would it be more sensible to pass all updates to the KDS side of things?  

Or route updates to Kafka if the feature is not in Accumulo?

Also, I think if we go this route, then we really, really ought to implement deduplication for Arrow, Bin, Density...

Reply:
all writes go to kafka first, including all updates. we could check the long-term store to see if there is an existing record, but then we'd probably have performance issues again as it would be similar to a modifying write without the lambda store.

The issue with arrow, bin and density is that we don't (usually) have enough info to deduplicate them in the client. so we'd have to either skip all server-side processing, or push down the list of fids to exclude (which may be large). I also think with those queries, it's likely to not affect the UI much if there are some duplicates (densities would be slightly off, bins/arrows would show duplicate points but they would just overlap each other and any lookup click would only return the one feature, tables and charts would be off though).